src/OrleansAzureUtils/AzureSilo.cs

```diff
@@ -150,29 +130,29 @@ public bool Start(string deploymentId, RoleInstance myRoleInstance, ClusterConfi
 
             myEntry = new SiloInstanceTableEntry
             {
-                DeploymentId = deploymentId,
-                Address = myEndpoint.IPEndpoint.Address.ToString(),
-                Port = myEndpoint.IPEndpoint.Port.ToString(CultureInfo.InvariantCulture),
+                DeploymentId = serviceRuntimeWrapper.DeploymentId,
+                Address = myEndpoint.Address.ToString(),
```

Comment:
because of line like this it is important NOT to use var in "var myEndpoint = serviceRuntimeWrapper.GetEndpoint(SiloEndpointConfigurationKeyName);"
How am I supposed to guess that you changed  serviceRuntimeWrapper.GetEndpoint to return IPEndpoint while it used to be RoleInstanceEndpoint ?

It is definitely not true that by looking at `var myEndpoint = serviceRuntimeWrapper.GetEndpoint(SiloEndpointConfigurationKeyName);` it obvious to deduce the type of myEndpoint .

Bottom line: don't use var there.